### PR TITLE
Replace deprecated url import

### LINKED ISSUE #41: `django.conf.urls.url` deprecated as of Django 3.1
The project currently makes use of `django.conf.urls.url`. As of Django 3.1 `django.conf.urls.url()` alias of `django.urls.re_path()` is deprecated.

To ensure future compatibility (especially with Django 4.0 which is about to be released), we should only rely on the alias for older Django versions. 

See also: 
https://docs.djangoproject.com/en/3.2/releases/3.1/#id2

### PR DESCRIPTION
**Pull request by tbrlpld**
The django.conf.urls.url() alias of django.urls.re_path() is
deprecated as of Django 3.1 and will be removed in 4.0 (which is
currently in alpha).

This commit changes the import to use the orignal alias for versions
before 3.1 and uses django.urls.re_path directly for newer versions.

See also:
https://docs.djangoproject.com/en/3.2/releases/3.1/#id2

Closes #41

### CODE REVIEW

File: generic_chooser/views.py
Comment (zerolab) [inline generic_chooser/views.py:27]: You should be able to import directly from `django.urls` since the beginning - https://github.com/django/django/blob/stable/2.0.x/django/urls/__init__.py

Comment (tbrlpld) [reply generic_chooser/views.py:27]: Good call. I did not even check that.

File: tests/urls.py
Comment (zerolab) [inline tests/urls.py:28]: nitpick: all of these are better of with `path()`

**Review by zerolab — APPROVED** [review]
LGTM

**Comment by tbrlpld** [comment]
@zerolab I do not have write access in the project. It would be highly appreciated if you could do me the honour and press the button 😅🙏

**Comment by zerolab** [comment]
Unofortunately, I do not have such access, so will leave it to the powers that be to receive all the honours 😂

**Comment by gasman** [comment]
Now released in 0.3.1!

**Comment by tbrlpld** [comment]
Perfect 🎉 Thanks a lot  @gasman

**Pull request merged**
